From a4b5f8b29e513d369d5a8f1787d11b7229285742 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 12 Mar 2021 13:13:37 +0100
Subject: [PATCH 050/118] Revert "bfq: expire bfqq if a higher priority class
 is waiting"

This reverts commit eb30cbc0c9feb1ea804ca47ec9193ae1b5fdb76f.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index a26b11fde..0c91925ba 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -4786,10 +4786,7 @@ static struct request *bfq_dispatch_rq_from_bfqq(struct bfq_data *bfqd,
 						 struct bfq_queue *bfqq)
 {
 	struct request *rq = bfqq->next_rq;
-	struct bfq_sched_data *sd = bfqq->entity.sched_data;
-	struct bfq_service_tree *st = sd->service_tree;
 	unsigned long service_to_charge;
-	int idx;
 
 	service_to_charge = bfq_serv_to_charge(rq, bfqq);
 
@@ -4826,17 +4823,6 @@ static struct request *bfq_dispatch_rq_from_bfqq(struct bfq_data *bfqd,
 	if (!(bfq_tot_busy_queues(bfqd) > 1 && bfq_class_idle(bfqq)))
 		goto return_rq;
 
-	/*
-	 * Expire bfqq, if other queues belong to higher priority
-	 * class are waiting for service.
-	 */
-	for (idx = bfqq->ioprio_class - 2; idx >= 0; idx--) {
-		if (!RB_EMPTY_ROOT(&(st + idx)->active))
-			break;
-	}
-	if (idx < 0)
-		goto return_rq;
-
 	bfq_bfqq_expire(bfqd, bfqq, false, BFQQE_BUDGET_EXHAUSTED);
 
 return_rq:
-- 
2.31.1.305.gd1b10fc6d8

